binaries/oli/src/config/mod.rs

```diff
@@ -0,0 +1,159 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::collections::HashMap;
+use std::fs;
+use std::path::Path;
+
+use anyhow::{anyhow, Result};
+use opendal::{services, Operator};
+use serde::Deserialize;
+use toml;
+
+type StringMap<T> = HashMap<String, T>;
+
+macro_rules! set_bucket {
+    ($m: expr, $bucket: ident) => {{
+        let mut opts = $m.clone().unwrap();
+        opts.insert("bucket".to_string(), $bucket.to_string());
+        opts
+    }};
+}
+
+#[derive(Deserialize, Default)]
+pub struct Config {
+    profiles: StringMap<Profile>,
+}
+
+#[derive(Deserialize, Default)]
+pub struct Profile {
+    // TODO: add more services
```

Comment:
Can we handle them in a more general way? Add seperate field for every service seems boring.